openpype/hosts/houdini/api/lib.py

```diff
@@ -804,6 +805,45 @@ def get_camera_from_container(container):
     return cameras[0]
 
 
+def get_current_context_template_data_with_asset_data():
+    """
```

Comment:
could we keep some of the old doc string ? 
```suggestion
    """Template data for template fill from current context

    Returns:
        Dict[str, Any] of the following tokens and their values
        Supported Tokens:
            - Regular Tokens
                - app
                - user
                - asset
                - parent
                - hierarchy
                - folder[name]
                - root[work, ...]
                - studio[code, name]
                - project[code, name]
                - task[type, name, short]

            - Context Specific Tokens
                - assetData[frameStart]
                - assetData[frameEnd]
                - assetData[handleStart]
                - assetData[handleEnd]
                - assetData[frameStartHandle]
                - assetData[frameEndHandle]
                - assetData[resolutionHeight]
                - assetData[resolutionWidth]
```

Reply:
It would have to be duplicated in 4 functions: `get_template_data`, `get_template_data_with_names`, `get_template_data_from_session`, `get_current_context_template_data`.
I would change the structure, something like:
```
Example output:
        {
            "project": {                        # Project data
                "name": "Commercial",           # - name
                "code": "cmr"                   # - code
            },
            "asset": "Buddy",                   # Asset name
            "hierarchy": "assets/characters",   # Asset parent hierarchy
            "parent": "characters",             # Asset parent name
            "folder": {                         # Folder data (AYON asset)
                "name": "Buddy"                 # - name
            },
            "task": {                           # Task data
                "name": "animation",            # - name
                "type": "Animation",            # - type
                "short": "anim"                 # - short name
            },
            "studio": {                         # Studio data (from settings)
                "name": "Perfect Video Makers", # - name
                "code": "PVM"                   # - code
            },
            "app": "maya"                       # Host name
        }
```

I don't think it is necessary, if you think it is, please create a PR.